src/rust/rust-proto/src/client_macros.rs

```diff
@@ -45,3 +47,38 @@ macro_rules! execute_client_rpc {
         }
     }};
 }
+
+/// This macro implements boilerplace code to connect to
+/// a gRPC service (and do retries if needed).
+/// Unfortunately, each $proto_client_type doesn't share traits, so a
+/// macro is the quickest solution.
+#[macro_export]
+macro_rules! get_proto_client {
+    (
+        $executor: ident,
+        $proto_client_type: ty,
+        $endpoint: ident,
+    ) => {{
+        {
+            let backoff = client_executor::strategy::FibonacciBackoff::from_millis(10);
```

Comment:
This is an excessive delay.. the 20th Fibonacci number is 4181 or 6765, depending on how you start counting.. if we're multiplying that by _ten_ here that's a 41s to 67s delay.

I think starting with a 10ms delay and backing off according to this Fibonacci strategy is fine, but we should
1. make sure each delay gets some random jitter added to it on the order of magnitude of the _previous_ step in the sequence and
2. make sure the total delay is capped at a reasonable value (e.g. 5s + jitter)

Reply:
I agree the numbers here are funky and in general I prefer exponential; I was just reusing the numbers colin gave me

Reply:
For a total timeout of 5000ms I'm doing
(fibonacci of 10 total) = 1 + 1 + 2 + 3 + 5 + 8 + 13 + 21 + 34 + 55 = 143
5000/143 = 35

so 35ms taken 10 times.

Reply:
What I mean by "capped at 5s" is the _longest_ timeout in the sequence should be 5s but once we hit that it shouldn't increase further. So it should go 35ms + O(1ms) jitter, 35ms + O(1ms) jitter, 70ms + O(1ms) jitter, 105ms + O(10ms) jitter ..., 5000ms + O(100ms) jitter, 5000ms + O(100ms) jitter, 5000ms + O(100ms) jitter...

Reply:
got it got it